src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -771,30 +772,202 @@ public class B
+                );
+
+            var method1ContainingType = method1.ContainingType; //A<T>
+            var method2ContainingType = method2.ContainingType; //A<T!>
+
+            VerifyEquality(method1ContainingType, method2ContainingType,
+                expectedDefault: true,
+                expectedIncludeNullability: false
+                );
+        }
+
+        [Fact]
+        [WorkItem(8195, "https://github.com/dotnet/roslyn/issues/38195")]
+        public void SemanticModel_SubstitutedEvent_Equality()
+        {
+            var source =
+@"
+#nullable enable
+public class A<T> 
+    where T : class //not neccesary, but makes it easier to reason about the resulting fields
+{
+    public event System.EventHandler<T> MyEvent;
+    public static void M(A<T> t)
+    {
+        _  = t.MyEvent;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,41): warning CS8618: Non-nullable event 'MyEvent' is uninitialized. Consider declaring the event as nullable.
+                //     public event System.EventHandler<T> MyEvent;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "MyEvent").WithArguments("event", "MyEvent").WithLocation(6, 41)
+                );
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+
+            var member1Syntax = (ClassDeclarationSyntax)root.DescendantNodes().First(sn => sn.Kind() == SyntaxKind.ClassDeclaration);
+            var member2Syntax = (IdentifierNameSyntax)root.DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.IdentifierName);
+
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var event1 = (IEventSymbol)((INamedTypeSymbol)model.GetDeclaredSymbol(member1Syntax)).GetMembers("MyEvent").Single(); // System.EventHandler<T!>! A<T>.MyEvent
+            var event2 = (IEventSymbol)model.GetSymbolInfo(member2Syntax).Symbol;                                                 // System.EventHandler<T!>! A<T!>.MyEvent
+
+            VerifyEquality(event1, event2,
+                expectedDefault: true,
+                expectedIncludeNullability: false
+                );
+
+            var event1Type = event1.Type; // System.EventHandler<T!>
+            var event2Type = event2.Type; // System.EventHandler<T!>
+
+            VerifyEquality(event1Type, event2Type,
+                expectedDefault: true,
+                expectedIncludeNullability: true
+                );
+
+            var event1ContainingType = event1.ContainingType; //A<T>
+            var event2ContainingType = event2.ContainingType; //A<T!>
+
+            VerifyEquality(event1ContainingType, event2ContainingType,
+                expectedDefault: true,
+                expectedIncludeNullability: false
+                );
+        }
+
+        private void VerifyEquality(ISymbol symbol1, ISymbol symbol2, bool expectedDefault, bool expectedIncludeNullability)
```

Comment:
`expectedDefault` seems to never be `false`. Consider either adding negative tests and adjusting hashcode verification accordingly, or removing the parameter.